### PR TITLE
test: add comprehensive test coverage for AccessControl contracts

### LINKED ISSUE #131: [Testing]: Add comprehensive test coverage for AccessControl contracts
The AccessControl contracts (`AccessControlFacet.sol` and `LibAccessControl.sol`) currently lack test coverage. These contracts provide role-based access control functionality, which is critical security infrastructure that must be thoroughly tested.

### PR DESCRIPTION
**Pull request by adamgall**
test: add comprehensive test coverage for AccessControl contracts (#131)

Implement complete test suites for LibAccessControl and AccessControlFacet
with test harnesses to expose internal library functions for thorough testing.

New files added:
- test/access/AccessControl/LibAccessControl.t.sol (454 lines, 37 tests)
- test/access/AccessControl/AccessControlFacet.t.sol (611 lines, 45 tests)
- test/access/AccessControl/harnesses/LibAccessControlHarness.sol
- test/access/AccessControl/harnesses/AccessControlFacetHarness.sol

Test coverage includes:
- Core RBAC functionality (grant, revoke, renounce, hasRole)
- Role admin hierarchies and custom role admins
- Access control enforcement with requireRole
- Storage slot verification and collision prevention
- Event emission for all role operations
- Edge cases (circular hierarchies, self-admin roles, zero addresses)
- Comprehensive fuzz tests for all operations
- Complex multi-level role hierarchy scenarios

Total: 82 new tests ensuring robust access control implementation
following patterns established by Owner and OwnerTwoSteps test suites.

Closes #131

### CODE REVIEW

File: test/access/AccessControl/LibAccessControl.t.sol [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+
+    function test_GasOptimization_HasRole() public {
+        harness.forceGrantRole(MINTER_ROLE, ALICE);
+
+        uint256 gasBefore = gasleft();
+        harness.hasRole(MINTER_ROLE, ALICE);
+        uint256 gasUsed = gasBefore - gasleft();
+
+        console2.log("Gas used for hasRole:", gasUsed);
+        assertEq(gasUsed, 1377, "hasRole gas cost should be exactly 1377");
```
Comment (maxnorm) [inline test/access/AccessControl/LibAccessControl.t.sol]: Using fix gas assumption, in test, break the coverage.

This gas value is based on the assumption of optimizer runs which the coverage action disable. The ` optimizer_runs` value is set to 20_000, but this value can also be change. This will also break the tests.

Upper limit seems a good way to test gas usage within a appropriate range while i agree that it's not the most accurate way.

**Comment by adamgall** [comment]
will fix these broken actions

File: test/access/AccessControl/LibAccessControl.t.sol (hunk 1 above)
Comment (adamgall) [reply test/access/AccessControl/LibAccessControl.t.sol]: best way IMO is to rip out these "gas tests" entirely, and utilize `forge snapshot` for tracking gas costs between commits / code.

Comment (adamgall) [reply test/access/AccessControl/LibAccessControl.t.sol]: https://github.com/Perfect-Abstractions/Compose/issues/134
